python/_brotli.cc

```diff
@@ -88,6 +87,37 @@ static int lgblock_convertor(PyObject *o, int *lgblock) {
   return 1;
 }
 
+static BROTLI_BOOL compress_stream(BrotliEncoderState* enc, BrotliEncoderOperation op,
+                                   std::vector<uint8_t>* output, uint8_t* input, size_t input_length) {
+  BROTLI_BOOL ok = BROTLI_TRUE;
+
+  while (ok) {
+    size_t available_in = input_length;
+    const uint8_t* next_in = input;
```

Comment:
available_in and next_in are reset at the beginning of the loop.
This will cause almost infinite loop, if output does not fit internal encoder buffer...